core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -278,8 +278,12 @@ object DumpLogSegments {
         }
       }.mkString("{", ",", "}")
 
-      val keyString = s"metadata::$groupId"
-      val valueString = s"$protocolType:${group.protocol}:${group.generationId}:$assignment"
+      val keyString = Json.encode(Map("metadata" -> groupId))
+      val valueString = Json.encode(Map(
+          "protocolType" -> protocolType,
+          "groupMetadata.protocol" -> group.protocol,
+          "groupMetadata.generationId" -> group.generationId,
+          "assignment" -> assignment))
```

Comment:
Should we jsonify the assignment too, or do you think it's ok as is?

Reply:
I guess it wouldn't hurt if we jsonify that too. Will update it.